include/swift/Frontend/DiagnosticVerifier.h

```diff
@@ -18,22 +18,124 @@
 #ifndef SWIFT_FRONTEND_DIAGNOSTIC_VERIFIER_H
 #define SWIFT_FRONTEND_DIAGNOSTIC_VERIFIER_H
 
+#include "swift/AST/DiagnosticConsumer.h"
 #include "swift/Basic/LLVM.h"
+#include "llvm/ADT/SmallSet.h"
+#include "llvm/ADT/StringSet.h"
 
 namespace swift {
-  class SourceManager;
-
-  /// Set up the specified source manager so that diagnostics are captured
-  /// instead of being printed.
-  void enableDiagnosticVerifier(SourceManager &SM);
-
-  /// Verify that captured diagnostics meet with the expectations of the source
-  /// files corresponding to the specified \p BufferIDs and tear down our
-  /// support for capturing and verifying diagnostics.
-  ///
-  /// This returns true if there are any mismatches found.
-  bool verifyDiagnostics(SourceManager &SM, ArrayRef<unsigned> BufferIDs,
-                         bool autoApplyFixes, bool ignoreUnknown);
+/// This class implements support for -verify mode in the compiler.  It
+/// buffers up diagnostics produced during compilation, then checks them
+/// against expected-error markers in the source file.
+class DiagnosticVerifier : public DiagnosticConsumer {
```

Comment:
Do we know why `DiagnosticVerifier` wasn't a `DiagnosticConsumer` before? I'm worried that this change might make the verifier not, for instance, compose with other diagnostic consumers the way it used to. Do `-serialize-diagnostics-path` and `-emit-fixits-path` in combination with `-verify` work the same way they used to? 

Separately: Why is `DiagnosticVerifier` fully defined in an `include/` header? The other diagnostic consumers only declare a function to create an instance and hide the rest of the implementation, and the old `-verify` implementation also only exposed a few functions to set it up. Is there an architectural difference that forces `DiagnosticVerifier` to be different, or was this simply an oversight?